### PR TITLE
Add Collect creator data from Europeana API 

### LINKED ISSUE #2834: Europeana script should collect the creator
## Problem

The Europeana script does not collect the creator data.

## Description

Europeana is an aggregator of high-quality media from many European GLAM institutions, and it is important for search relevancy to collect all of the relevant data. Creator is not currently not collected at all.

The creators are available in the `dcCreator` field within the returned data, here's an example with our sample data: 

https://github.com/WordPress/openverse/blob/aedc9c16ce5ed11709e2b6f0b42bad77ea1eb19b/catalog/tests/dags/providers/provider_api_scripts/resources/europeana/europeana_example.json#L6882

Here's what it looks like within a real return data:

```python
{'completeness': 8,
 'country': ['Hungary'],
 'dataProvider': ['Mezőkövesd Public Treasures Collection Nonprofit Ltd.'],
 'dcCreator': ['Hajdu Ráfis János Mezőgazdasági Gépmúzeum Mezőkövesd'],
 'dcCreatorLangAware': {'def': ['Hajdu Ráfis János Mezőgazdasági Gépmúzeum Mezőkövesd']},
 'dcDescription': ['MÁV Gépgyár (Budapest)                             Állapota: keresztfeje, vízszivattyúja, hajtórúdjai hiányoznak. Füstkamrája elrozsdásodott. Kiállításra alkalmassá tehető.'],
 'dcDescriptionLangAware': {'hu': ['MÁV Gépgyár (Budapest)                             Állapota: keresztfeje, vízszivattyúja, hajtórúdjai hiányoznak. Füstkamrája elrozsdásodott. Kiállításra alkalmassá tehető.']},
 'dcLanguage': ['zxx', 'ZXX'],
 'dcLanguageLangAware': {'def': ['zxx', 'ZXX']},
 'dcSubjectLangAware': {'hu': ['Agráltörténet', 'színes kép']},
 'dcTitleLangAware': {'hu': ['MÁV Gépgyár  által készített gőzüzemű lokomobil']},
 'dcTypeLangAware': {'def': ['kép']},
 'dctermsSpatial': ['Mezőkövesd'],
 'edmDatasetName': ['2048128_Ag_HU_MaNDA_OAI'],
 'edmIsShownAt': ['https://mandadb.hu/tetel/99947/MAV_Gepgyar__altal_keszitett_gozuzemu_lokomobil'],
 'edmIsShownBy': ['https://mandadb.hu/common/file-servlet/document/175231/default/doc_url/200021_3.JPG'],
 'edmPreview': ['https://api.europeana.eu/thumbnail/v2/url.json?uri=https%3A%2F%2Fmandadb.hu%2Fmandadb%2Fwebimage%2F9%2F5%2F6%2F5%2F3%2Fwimage%2F200021_2.jpg&type=IMAGE'],
 'europeanaCollectionName': ['2048128_Ag_HU_MaNDA_OAI'],
 'europeanaCompleteness': 8,
 'guid': 'https://www.europeana.eu/item/2048128/99947?utm_source=api&utm_medium=api&utm_campaign=dialialika',
 'id': '/2048128/99947',
 'index': 0,
 'language': ['hu'],
 'link': 'https://api.europeana.eu/record/2048128/99947.json?wskey=dialialika',
 'organizations': ['http://data.europeana.eu/organization/1482250000004509149',
  'http://data.europeana.eu/organization/1482250000003772998'],
 'previewNoDistribute': False,
 'provider': ['Forum Hungaricum Non-profit Ltd.'],
 'rights': ['http://creativecommons.org/licenses/by-nc-nd/4.0/'],
 'score': 1.0,
 'timestamp': 1716914248926,
 'timestamp_created': '2024-05-28T07:50:08.374Z',
 'timestamp_created_epoch': 1716882608374,
 'timestamp_update': '2024-05-28T07:50:08.374Z',
 'timestamp_update_epoch': 1716882608374,
 'title': ['MÁV Gépgyár  által készített gőzüzemű lokomobil'],
 'type': 'IMAGE',
 'ugc': [False]}
```

It looks like there could be a list of creators, we should probably join them together with commas (e.g. `", ".join(item_data.get("dcCreator", ""))`).

In order to accomplish this, we'll need to modify the [Europeana provider ingestion script](https://github.com/WordPress/openverse/blob/7f4fb7c91363747505d72a63dc7509617d16acfc/catalog/dags/providers/provider_api_scripts/europeana.py). We should add a new function to the `EuropeanaRecordBuilder` class to retrieve this value. A good example to use would be [`_get_filesize`](https://github.com/WordPress/openverse/blob/7f4fb7c91363747505d72a63dc7509617d16acfc/catalog/dags/providers/provider_api_scripts/europeana.py#L156-L157). We'll then need to capture this information in `get_record_data`, by adding it to the dictionary with the `"creator"` key here:

https://github.com/WordPress/openverse/blob/7f4fb7c91363747505d72a63dc7509617d16acfc/catalog/dags/providers/provider_api_scripts/europeana.py#L73

We'll also need to add tests for this function and update any other Europeana tests that might be affected. An example test for `_get_filesize` can be [found here](https://github.com/WordPress/openverse/blob/33f9f7c54c2a27593716ce429b2b4b42f8df6e22/catalog/tests/dags/providers/provider_api_scripts/test_europeana.py#L299-L308).

### PR DESCRIPTION
**Pull request by dryruffian**
## Fixes

Fixes #2834  by @obulat 

## Description

Added a New method `_get_creator` in class `EuropeanaRecordBuilder` to collect cerator data form `dcCreator` field, which contains the creator information, is now extracted and included in the record data.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dryruffian** [comment]
Yo @obulat, can you please review this PR? It adds a new method to the EuropeanaRecordBuilder class that collects creator data from the dcCreator field.

File: catalog/dags/providers/provider_api_scripts/europeana.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def get_record_data(self, data: dict) -> dict | None:
                 "foreign_identifier": self._get_foreign_identifier(data),
                 "meta_data": self._get_meta_data_dict(data),
                 "title": self._get_title(data),
+                "creator": self._get_creator(item_data),
```
Comment (obulat) [inline catalog/dags/providers/provider_api_scripts/europeana.py:71]:
```suggestion
                "creator": self._get_creator(data),
```

I tested this PR using October 1st as the date. The `dcCreator` is available in `data` dictionary, not `item_data`.

File: catalog/dags/providers/provider_api_scripts/europeana.py [hunk 2]
```diff
@@ -157,6 +158,12 @@ def _get_filetype(self, item_data: dict) -> str:
     def _get_filesize(self, item_data: dict) -> int:
         return item_data.get("ebucoreFileByteSize")
 
+    def _get_creator(self, item_data: dict) -> str | None:
+        creators = item_data.get("dcCreator", [])
+        if not creators:
+            return None
+        return creators if isinstance(creators, str) else ", ".join(creators)
```
Comment (obulat) [inline catalog/dags/providers/provider_api_scripts/europeana.py:165]: I think it's better to check if the creators is an instance of list or not:
```py
if isinstance(creators, list):
    return ", ".join(creators)
elif isinstance(creators, str):
    return creators
return None
```

File: catalog/tests/dags/providers/provider_api_scripts/test_europeana.py [hunk 3]
```diff
@@ -284,6 +284,33 @@ def test_get_image_dimensions(item_data, expected, record_builder):
     assert record_builder._get_image_dimensions(item_data) == expected
 
 
+@pytest.mark.parametrize(
+    "item_data, expected",
+    [
+        # Single creator in a list
+        pytest.param({"dcCreator": ["Chandler"]}, "Chandler", id="single_creator"),
```
Comment (obulat) [inline catalog/tests/dags/providers/provider_api_scripts/test_europeana.py:291]: 😆

**Review by obulat — CHANGES_REQUESTED** [review]
This PR is almost there, @dryruffian! There are two points that need to be fixed to make it working correctly.

File: catalog/dags/providers/provider_api_scripts/europeana.py (hunk 2 above)
Comment (dryruffian) [reply catalog/dags/providers/provider_api_scripts/europeana.py:165]: thank's @obulat for the suggestion I have updated the code to check creators is an instance of list or not. I have also added also tests for some edge cases

**Comment by dryruffian** [comment]
Hi, @obulat can you assist me with the codespell issue

**Review by krysal — COMMENTED** [review]
Jumping in to leave some observations!

**Comment by dryruffian** [comment]
Hi @krysal, 
I have applied the suggested changes can you review it now

File: catalog/dags/providers/provider_api_scripts/europeana.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ def _get_creator(self, data: dict) -> str | None:
         if isinstance(creators, list):
             if not creators:  # Explicitly handle empty list
```
Comment (obulat) [inline catalog/dags/providers/provider_api_scripts/europeana.py:165]: It would be better to move the `if not creators` condition higher, so we return `None` both for an empty list and an empty string (`""`).

File: catalog/dags/providers/provider_api_scripts/europeana.py [hunk 5]
```diff
@@ -157,6 +158,17 @@ def _get_filetype(self, item_data: dict) -> str:
     def _get_filesize(self, item_data: dict) -> int:
         return item_data.get("ebucoreFileByteSize")
 
+    @staticmethod
+    def _get_creator(self, data: dict) -> str | None:
+        creators = data.get("dcCreator", [])
+        if isinstance(creators, list):
+            if not creators:  # Explicitly handle empty list
+                return None
+            return ",".join(creators)
```
Comment (obulat) [inline catalog/dags/providers/provider_api_scripts/europeana.py:167]:
```suggestion
            return ", ".join(creators)
```
The first version, with the space, was better.

**Review by obulat — CHANGES_REQUESTED** [review]
This is almost there! Just 2 tiny changes needed

**Comment by obulat** [comment]
For fixing the codespell linting issue, I think instead of adding the ignore text, it's better to add the test resources folder to the ignore array:
https://github.com/WordPress/openverse/blob/3078831031c4950d943d9d1c02eb850573de3ec9/.pre-commit-config.yaml#L75

File: catalog/tests/dags/providers/provider_api_scripts/test_europeana.py [hunk 6]
```diff
@@ -186,6 +186,8 @@ def test_record_builder_get_record_data(ingester, record_builder):
         "description": "Sello en seco: España artística y monumental.",
     }
 
+    expected_creator = "http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent2,http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent3,http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent4"
```
Comment (obulat) [inline catalog/tests/dags/providers/provider_api_scripts/test_europeana.py:189]:
```suggestion
    expected_creator = "http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent2,http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent3,http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent4"  # codespell:ignore
```

File: .codespell/ignore_lines.txt [hunk 7]
```diff
@@ -20,3 +20,10 @@
 ;; packages/js/eslint-plugin/configs/vue.ts
 ;; `te` gets matched with `the` and others
 const i18nDestructureRules = ["t", "tc", "te", "td", "d", "n"].map(
+
+;; catalog/tests/dags/providers/provider_api_scripts/test_europeana.py
+;; lod refers to a legitimate part of a URL, likely meaning 
+;;"Linked Open Data" (LOD), which is commonly used in metadata contexts
+"http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent2, "
+"http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent3, "
+"http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent4"
```
Comment (obulat) [inline .codespell/ignore_lines.txt:29]:
```suggestion

```
We can use an inline comment to ignore this error

File: catalog/tests/dags/providers/provider_api_scripts/test_europeana.py [hunk 8]
```diff
@@ -284,6 +287,33 @@ def test_get_image_dimensions(item_data, expected, record_builder):
...
 
+@pytest.mark.parametrize(
+    "item_data, expected",
+    [
+        # Single creator in a list
+        pytest.param({"dcCreator": ["Chandler"]}, "Chandler", id="single_creator"),
+        # Multiple creators in a list
+        pytest.param(
+            {"dcCreator": ["Chandler", "Joey"]},
+            "Chandler,Joey",
```
Comment (obulat) [inline catalog/tests/dags/providers/provider_api_scripts/test_europeana.py:298]:
```suggestion
            "Chandler, Joey",
```

File: catalog/tests/dags/providers/provider_api_scripts/test_europeana.py [hunk 9]
```diff
@@ -186,6 +186,8 @@ def test_record_builder_get_record_data(ingester, record_builder):
         "description": "Sello en seco: España artística y monumental.",
     }
 
+    expected_creator = "http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent2,http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent3,http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent4"  # codespell:ignore
```
Comment (obulat) [inline catalog/tests/dags/providers/provider_api_scripts/test_europeana.py:189]:
```suggestion
    expected_creator = "http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent2, http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent3, http://hispana.mcu.es/lod/oai:bibliotecadigital.jcyl.es:26229#ent4"  # codespell:ignore
```

**Review by obulat — APPROVED** [review]
I added the last bit of change requests to this PR, and now the CI is passing.

Thank you for your contribution, @dryruffian! Europeana has very high-quality media, and now Openverse will have better data for these media items, and they will be easier to find ✨
